src/ex.rs

```diff
@@ -61,13 +62,32 @@ fn froml_path(ex_name: &str, name: &str, vers: &str) -> PathBuf {
 }
 
 #[derive(Serialize, Deserialize)]
+pub struct SerializableExperiment {
```

Comment:
I assume you added because you added a mutex to `Experiment`, but I believe `serde` is happy to serialize that? https://docs.rs/serde/1.0.36/serde/trait.Serialize.html?search=#impl-Serialize-86

Then I think you can get rid of `ShasMap::new`

Reply:
This is not due to the mutex. `ShasMap` needs the experiment name in order to be created, and I can't get that during deserialization (as far as I know). I need first to deserialize the data (in this extra struct) and _then_ convert it into an experiment creating the `ShasMap`.

Sure, I could implement a custom deserializer, but that's way uglier in my opinion.

Reply:
Other than that, crater currently serializes the shas in a separate `shas.json` file. I could add a `#[serde(skip)]` to fix that, but the other problem would still be present.